src/istio/control/tcp/attributes_builder.cc

```diff
@@ -51,6 +51,9 @@ void AttributesBuilder::ExtractCheckAttributes(CheckData* check_data) {
   builder.AddBool(utils::AttributeName::kConnectionMtls,
                   check_data->IsMutualTLS());
 
+  builder.AddString(utils::AttributeName::kConnectionRequestedServerName,
```

Comment:
can you please check whether the string is empty first? we normally don't send unset values to mixer.